modules/apps/frontend-js/frontend-js-a11y-web/src/main/resources/META-INF/resources/useFilteredViolations.tsx

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import React, {Dispatch, createContext, useContext, useReducer} from 'react';
+
+import type {ImpactValue, Result} from 'axe-core';
+
+export const TYPES = {
+	CATEGORY_ADD: 'CATEGORY_ADD',
+	CATEGORY_REMOVE: 'CATEGORY_REMOVE',
+	IMPACT_ADD: 'IMPACT_ADD',
+	IMPACT_REMOVE: 'IMPACT_REMOVE',
+} as const;
+
+type TAction = {
+	payload: {value: string};
+	type: 'CATEGORY_ADD' | 'CATEGORY_REMOVE' | 'IMPACT_ADD' | 'IMPACT_REMOVE';
+};
+
+type TState = {
+	filteredViolations: Array<Result>;
+	selectedCategories: Array<String> | [];
+	selectedImpact: Array<ImpactValue> | [];
+	violations: Array<Result>;
+};
+
+const INITIAL_STATE: TState = {
+	filteredViolations: [],
+	selectedCategories: [],
+	selectedImpact: [],
+	violations: [],
+};
+
+function filterByCategories(
+	receivedTags: Array<String>,
+	selectedCategories: Array<String>
+) {
+	return selectedCategories.some((category) =>
+		receivedTags.includes(category)
+	);
+}
+
+function filterByImpact(
+	receivedImpact: ImpactValue,
+	selectedImpact: Array<ImpactValue>
+) {
+	return selectedImpact.includes(receivedImpact);
+}
+
+function addItem(list: Array<any>, value: any) {
+	return [...list, value];
+}
```

Comment:
I would avoid creating a function just to do this, it is a very simple thing that may not help, if you are thinking about lines, this seems unnecessary both for adding more lines and for creating a new function, doing all of that inline we have some gains will have the same effect of understanding.